doc/guides/writing-tests.md

```diff
@@ -223,6 +223,34 @@ countdown.dec();
 countdown.dec(); // The countdown callback will be invoked now.
 ```
 
+#### Testing promises
+
+When writing tests involving promises, either make sure that the
+`onFulfilled` or the `onRejected` handler is wrapped in
+`common.mustCall()` or `common.mustNotCall` accordingly, or
+call `common.crashOnUnhandledRejection()` in the top level of the
+test to make sure that unhandled rejections would result in a test
+failure. For example:
+
+```javascript
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs').promises;
+
+// Use `common.crashOnUnhandledRejection()` to make sure unhandled rejections
+// will fail the test
+common.crashOnUnhandledRejection();
+
+// Or, wrap the `onRejected` handler in `common.mustNotCall()`
+fs.writeFile('test-file', 'test').catch(common.mustNotCall());
+
+// Or, wrap the `onFulfilled` handler in `common.mustCall()`
+fs.readFile('test-file').then(
+  common.mustCall(
+    (content) => assert.strictEqual(content.toString(), 'test')
+  ));
```

Comment:
It's an example, but I think it makes sense to add `catch()` to handle the case where the assertion fails.

Reply:
@Ipinca that is demonstrated in the first example...also the `onFullfilled` and the `onRejected` handler are mutually exclusive so `common.mustCall` should be able to catch that.

Reply:
@lpinca I see what you mean now, the `catch` needs to handle possible failures if the `onFulfilled` handler is not empty. Thanks for catching that!